File: task1_compress.py
def compress(s):
    result = ""
    i = 0
    while i < len(s):
        c = s[i]
        count = 0
        while i < len(s) and s[i] == c:
            count += 1
            i += 1
        result += c + str(count)
    return result if len(result) < len(s) else s
 
s = input("Введите строку: ")
print("Результат:", compress(s))

File: task1_compress.js
const readline = require("readline");
const rl = readline.createInterface({ input: process.stdin });
 
function compress(s) {
    let result = "";
    let i = 0;
    while (i < s.length) {
        const c = s[i];
        let count = 0;
        while (i < s.length && s[i] === c) {
            count++;
            i++;
        }
        result += c + count;
    }
    return result.length < s.length ? result : s;
}
 
rl.question("Введите строку: ", (s) => {
    console.log("Результат:", compress(s));
    rl.close();
});
